### PR TITLE
Improve manual connection

### LINKED ISSUE #612: devicemanager: fix manual connection
Manual connection is apparently not working.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #612
Needs: https://github.com/bluerobotics/ping-cpp/pull/8

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -67,6 +70,11 @@ int main(int argc, char *argv[])
         "AbstractLinkNamespace", 1, 0, "AbstractLinkNamespace", "This is another enum."
     );
 
+    qmlRegisterUncreatableMetaObject(
+        PingEnumNamespace::staticMetaObject,
+        "PingEnumNamespace", 1, 0, "PingEnumNamespace", "This is another enum."
```
Comment (tcanabrava) [inline src/main.cpp]: "this is another enum" is a bad description. This description is used thru IDE's to give us hints about what the code is.

File: qml/ConnectionMenu.qml [hunk 2]
```diff
@@ -81,39 +82,54 @@ Item {
...
+                text: "Device:"
                 enabled: true
                 color: Material.primary
             }
 
             ComboBox {
+                id: deviceCB
+                Layout.fillWidth: true
+                textRole: "name"
+                //TODO: This model should be provided by someone else and not to be hardcoded here
```
Comment (tcanabrava) [inline qml/ConnectionMenu.qml]: you can perhaps use the QMetaType information of the menu to retrieve the values, that are generated by Q_ENUM to fill a list model. but that can be in another patch.

File: qml/ConnectionMenu.qml [hunk 3]
```diff
@@ -81,39 +82,54 @@ Item {
...
+                    ListElement {
+                        name: "Ping360"
+                        deviceId: PingEnumNamespace.PingDeviceType.PING360
+                    }
+                }
+            }
+
+            Text {
+                text: "Communication:"
+                enabled: true
```
Comment (tcanabrava) [inline qml/ConnectionMenu.qml]: why "text" needs to be enabled?

File: qml/ConnectionMenu.qml [hunk 4]
```diff
@@ -197,9 +203,29 @@ Item {
...
-                    onEditingFinished: {
-                        connect(AbstractLinkNamespace.Udp, udpIp.text, text)
+                }
+            }
+
+            Button {
+                text: "Connect"
+                Layout.fillWidth: true
+                Layout.columnSpan: 5
+                onClicked: {
```
Comment (tcanabrava) [inline qml/ConnectionMenu.qml]: maybe this funcitons should go to c++ ?

File: qml/ConnectionMenu.qml [hunk 5]
```diff
@@ -197,9 +203,29 @@ Item {
                     text: "1234"
                     Layout.columnSpan:  2
                     Layout.fillWidth: true
-                    onEditingFinished: {
-                        connect(AbstractLinkNamespace.Udp, udpIp.text, text)
```
Comment (tcanabrava) [inline qml/ConnectionMenu.qml]: unrelated to the patch, but 'connect' is a very bad name for anything that deals with Qt.

File: qml/ConnectionMenu.qml (hunk 2 above)
Comment (patrickelectric) [reply qml/ConnectionMenu.qml]: The problem here is not the enum, but the map between string and enum.

File: qml/ConnectionMenu.qml (hunk 4 above)
Comment (patrickelectric) [reply qml/ConnectionMenu.qml]: There is a C++ function in the end to do the connection, this function only gets the necessary information for each QML control.

File: qml/ConnectionMenu.qml (hunk 5 above)
Comment (patrickelectric) [reply qml/ConnectionMenu.qml]: Function removed.

**Comment by patrickelectric** [comment]
@Williangalvani 	please test when possible.

File: qml/ConnectionMenu.qml [hunk 6]
```diff
@@ -205,9 +192,29 @@ Item {
...
                         top: 65535
                     }
-                    onEditingFinished: {
-                        connect(AbstractLinkNamespace.Udp, udpIp.text, text)
+                }
+            }
+
+            Button {
+                text: "Connect"
+                Layout.fillWidth: true
```
Comment (Williangalvani) [inline qml/ConnectionMenu.qml]: Can we do something like this to block connecting to invalid addreses?
```suggestion
                Layout.fillWidth: true
                // Disable if using an invalid UDP address
                enabled: conntype.currentIndex === 0 || udpIp.isValid
```
Can we get something like this to forbid connections to bad addresses?
